src/KeyVault/KeyVault.Test/Scripts/ControlPlane/KeyVaultManagementTests.ps1

```diff
@@ -836,4 +842,46 @@ function Test-NetworkRuleSet
 	{
 		Remove-AzResourceGroup -Name $resourceGroupName -Force
 	}
+}
+
+function Test-UpdateKeyVault {
+    $resourceGroupName = getAssetName
+    $resourceGroupLocation = Get-Location "Microsoft.Resources" "resourceGroups" "westus"
+    $vaultLocation = Get-Location "Microsoft.KeyVault" "vaults" "westus"
+
+    try {
+        $rg = New-AzResourceGroup -Name $resourceGroupName -Location $resourceGroupLocation
+        $vault = New-AzKeyVault -VaultName (getAssetName) -ResourceGroupName $resourceGroupName -Location $vaultLocation
+        Assert-True { $vault.EnableSoftDelete -ne $true } "1. EnableSoftDelete should not be true"
+        Assert-True { $vault.EnablePurgeProtection -ne $true } "1. EnablePurgeProtection should not be true"
+
+        # Enable soft delete first
+        $vault = $vault | Update-AzKeyVault -EnableSoftDelete
+        Assert-True { $vault.EnableSoftDelete } "2. EnableSoftDelete should be true"
+        Assert-True { $vault.EnablePurgeProtection -ne $true } "2. EnablePurgeProtection should not be true"
+        # Assert-AreEqual 90 $vault.SoftDeleteRetentionInDays "2. SoftDeleteRetentionInDays should default to 90"
+
+        # Enable again
+        $vault = $vault | Update-AzKeyVault -EnableSoftDelete
+        Assert-True { $vault.EnableSoftDelete } "2.5. EnableSoftDelete should be true"
+        Assert-True { $vault.EnablePurgeProtection -ne $true } "2.5. EnablePurgeProtection should not be true"
+        
+        # Then enable purge protection
+        $vault = $vault | Update-AzKeyVault -EnablePurgeProtection
+        Assert-True { $vault.EnableSoftDelete } "3. EnableSoftDelete should be true"
+        Assert-True { $vault.EnablePurgeProtection } "3. EnablePurgeProtection should be true"
+
+        # Enable both together
+        $vault = New-AzKeyVault -VaultName (getAssetName) -ResourceGroupName $resourceGroupName -Location $vaultLocation
+        $vault = $vault | Update-AzKeyVault -EnableSoftDelete -EnablePurgeProtection
+        Assert-True { $vault.EnableSoftDelete } "4. EnableSoftDelete should be true"
+        Assert-True { $vault.EnablePurgeProtection } "4. EnablePurgeProtection should be true"
+        
+        # # Only enable purge protection (TODO: uncomment this assert after keyvault team deploys their fix)
```

Comment:
You may notice I checked in some commented code about retention, because the current .net sdk does not support it.
I will submit another PR to update .net sdk and uncomment these code.